### ISSUE TITLE
no function clause matching in Joken.Signer.rs/2

### ISSUE THREAD

**Issue by ItsRaWithTheH** [bryanjos/joken#169]
In a  Phoenix App. trying to use Joken.Plug in the router gives me an error at runtime

```
Request: POST /graphiql
** (exit) an exception was raised:
    ** (FunctionClauseError) no function clause matching in Joken.Signer.rs/2
        lib/joken/signer.ex:78: Joken.Signer.rs("RS256", "my_secret_key")
        (blog_phoenix) web/router.ex:20: BlogPhoenix.Router.verify_function/0
        lib/joken/plug.ex:174: Joken.Plug.parse_auth/2
        (blog_phoenix) web/router.ex:5: BlogPhoenix.Router.graphql/2
        (blog_phoenix) web/router.ex:1: BlogPhoenix.Router.match_route/4
        (blog_phoenix) web/router.ex:1: BlogPhoenix.Router.do_call/2
        (blog_phoenix) lib/blog_phoenix/endpoint.ex:1: BlogPhoenix.Endpoint.phoenix_pipeline/1
        (blog_phoenix) lib/plug/debugger.ex:123: BlogPhoenix.Endpoint."call (overridable 3)"/2
        (blog_phoenix) lib/blog_phoenix/endpoint.ex:1: BlogPhoenix.Endpoint.call/2
        (plug) lib/plug/adapters/cowboy/handler.ex:15: Plug.Adapters.Cowboy.Handler.upgrade/4
        :cowboy_protocol.execute/4
```

This is the router code:
```
defmodule BlogPhoenix.Router do
  use Phoenix.Router
  import Joken

  pipeline :graphql do
    plug Joken.Plug, verify: &BlogPhoenix.Router.verify_function/0, on_error: &BlogPhoenix.Router.error_logging/2
    plug BlogPhoenix.Context
  end

  scope "/" do
    pipe_through :graphql

    get "/graphiql", Absinthe.Plug.GraphiQL, schema: BlogPhoenix.Schema
    post "/graphiql", Absinthe.Plug.GraphiQL, schema: BlogPhoenix.Schema
    forward "/graphql", Absinthe.Plug, schema: BlogPhoenix.Schema
  end

  def verify_function() do
    %Joken.Token{}
    |> Joken.with_signer(rs256("my_secret_key"))
  end

  def error_logging(conn, message) do
    IO.puts("got an error")
    {conn, message}
  end
end
```

I am currently using joken 1.5.0

I cannot seem to see what is wrong as the Joken.Signer.rs convenience function does indeed take 2 parameters

**Comment by ItsRaWithTheH** [comment]
Update. My issue was trying to use a RS256 token which doesnt just work with a string secret. I have changed to use hs256 token

however i still get a "Could not verify token" response even though i have the hs256 secret correct (tested with jwt.io)

**Comment by cs-victor-nascimento** [comment]
Hi @ItsRaWithTheH! 

Well, your second issue seems a bit strange... we will need more info to understand why it is failing. 

Can you provide us with more info? This message might indicate a claim is not valid.

**Comment by aphillipo** [comment]
Yes, I’m sure it would help to provide the JWT, both encoded and unencoded so that people can attept to reproduce your issue.

**Comment by bryanjos** [comment]
I think it is because joken assumes the secret is not base64 encoded and jwt.io's demo has the secret as encoded. There is a comment on a closed issue [here](https://github.com/bryanjos/joken/issues/126#issuecomment-220327039) about it

**Comment by ItsRaWithTheH** [comment]
I am using an auth0 generated token for an example application
eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJuYW1lIjoiUmFodWwgTWFoaWRhIiwibmlja25hbWUiOiJJdHNSYVdpdGhUaGVIIiwiaXNzIjoiaHR0cHM6Ly9pdHNyYXdpdGh0aGVoLmF1dGgwLmNvbS8iLCJzdWIiOiJnaXRodWJ8NTM5MzU2MSIsImF1ZCI6Im1BTzA5dUE5ejViOUJjTEVtczJ2YWNLRFowNlpmSkhOIiwiZXhwIjoxNDk5ODU0MTc1LCJpYXQiOjE0OTk4MTgxNzV9.ha24ogf-3oQhrh-laLIpbMc2-bB-UMjK2MccKK_dy4s

according to Auth0 settings, my client secret is not base64 encoded (its an option that i have left unchecked for now), and using my secret in jwt.io correctly decodes it (see screenshot) https://drive.google.com/open?id=0B7PTiKI7FL3vRnVLQ29VZlZQSzQ

**Comment by cs-victor-nascimento** [comment]
@ItsRaWithTheH I tried to decode your token with hs256 using both "secret" (jwt.io default) and "my_secret_key". Obvisouly you've edited it for posting here and that is perfectly ok. 

Not sure how you are generating your token but I can see that you have iat and exp claims. Be sure they are being properly validated. Try adding some validations to your verify function ([here](https://github.com/bryanjos/joken/blob/master/test/plug_test.exs#L80) is an example validating a subject claim).

**Comment by victorolinasc** [comment]
Since we've just released 2.0.0-rc0 that was a complete rewrite and since this did not have any activity, I'll close this.

If the issue persists, please re-open. Thanks!
